sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs

```diff
@@ -0,0 +1,434 @@
+
+            if (matchesAutomaticEndpointsSpec || matchesExplicitEndpointsSpec)
+            {
+                if (matchesAutomaticEndpointsSpec && !settings.ContainsKey(TableConstants.ConnectionStrings.DefaultEndpointsProtocolSetting))
+                {
+                    settings.Add(TableConstants.ConnectionStrings.DefaultEndpointsProtocolSetting, "https");
+                }
+
+                var tableSecondaryEndpoint = settingOrDefault(TableConstants.ConnectionStrings.TableSecondaryEndpointSetting);
+
+                // if secondary is specified, primary must also be specified
+
+                static bool s_isValidEndpointPair(string primary, string secondary) =>
+                        !string.IsNullOrWhiteSpace(primary)
+                        || /* primary is null, and... */ string.IsNullOrWhiteSpace(secondary);
+
+                (Uri, Uri) createStorageUri(string primary, string secondary, string sasToken, Func<IDictionary<string, string>, (Uri, Uri)> factory)
+                {
+                    return
+                        !string.IsNullOrWhiteSpace(secondary) && !string.IsNullOrWhiteSpace(primary)
+                            ? (CreateUri(primary, sasToken), CreateUri(secondary, sasToken))
+                        : !string.IsNullOrWhiteSpace(primary)
+                            ? (CreateUri(primary, sasToken), default)
+                        : matchesAutomaticEndpointsSpec && factory != null
+                            ? factory(settings)
+                        : (default, default);
+
+                    static Uri CreateUri(string endpoint, string sasToken)
+                    {
+                        var builder = new UriBuilder(endpoint);
+                        if (!string.IsNullOrEmpty(builder.Query))
+                        {
+                            builder.Query += "&" + sasToken;
+                        }
+                        else
+                        {
+                            builder.Query = sasToken;
+                        }
+                        return builder.Uri;
+                    }
+                }
+
+                if (s_isValidEndpointPair(tableEndpoint, tableSecondaryEndpoint))
+                {
+                    accountInformation =
+                        new TableConnectionString(
+                            GetCredentials(settings),
+                            tableStorageUri: createStorageUri(tableEndpoint, tableSecondaryEndpoint, sasToken ?? string.Empty, ConstructTableEndpoint)
+                            )
+                        {
+                            EndpointSuffix = settingOrDefault(TableConstants.ConnectionStrings.EndpointSuffixSetting),
+                            Settings = settings
+                        };
+
+                    accountInformation._accountName = settingOrDefault(TableConstants.ConnectionStrings.AccountNameSetting);// ?? accountInformation.TableStorageUri.PrimaryUri.Host.Split(new[] { '.' })[0];
```

Comment:
Not sure what this comment tries to say.

Reply:
I didn't intend to leave that in - thanks